src/CSnakes.Runtime/CPython/Buffer.cs

```diff
@@ -0,0 +1,60 @@
+﻿using CSnakes.Runtime.Python;
+using System.Runtime.InteropServices;
+
+namespace CSnakes.Runtime.CPython;
+internal unsafe partial class CPythonAPI
+{
+    private const int PyBUF_SIMPLE = 0;
+    private const int PyBUF_WRITABLE = 0x0001;
+    private const int PyBUF_FORMAT = 0x0004;
+    private const int PyBUF_ND = 0x0008;
+    private const int PyBUF_STRIDES = (0x0010 | PyBUF_ND);
+    private const int PyBUF_C_CONTIGUOUS = (0x0020 | PyBUF_STRIDES);
+    private const int PyBUF_F_CONTIGUOUS = (0x0040 | PyBUF_STRIDES);
+    private const int PyBUF_ANY_CONTIGUOUS = (0x0080 | PyBUF_STRIDES);
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct Py_buffer
+    {
+        public IntPtr buf;
+        public IntPtr obj;
+        public nint len;
+        public Int32 @readonly;
```

Comment:
I get the why here, but it is generally better to use the keyword.

```suggestion
        public int @readonly;
```